Segunda parte

DentVega opened the pull request:
- Añadido del boton para eliminar una tarea
- Añadido del boton para editar una tarea

DentVega committed:
- Añadido del boton para eliminar una tarea
- Añadido del boton para editar una tarea

DentVega committed:
- modificacion de editar una tarea

combimauri commented on src/app/app.component.html:
```diff
@@ -6,8 +6,16 @@ <h1>tareas</h1>
+      <li *ngFor="let tarea of listaTareas; index as i">
         <label>{{ tarea }}</label>
+        <label>
+         <a class="btn  btn-red" (click)="eliminarTarea(i)">
```
```suggestion
         <a class="btn btn-red" (click)="eliminarTarea(i)">
```

combimauri commented on src/app/app.component.html:
```diff
@@ -6,8 +6,16 @@ <h1>tareas</h1>
   <section id="main" class="main">
     <ul id="todo-list" class="todo-list">
-      <li *ngFor="let tarea of listaTareas">
+      <li *ngFor="let tarea of listaTareas; index as i">
```
En vez de **i**, tal vez podrías usar algo como **indice**
```
      <li *ngFor="let tarea of listaTareas; index as indice">
```

combimauri commented on src/app/app.component.ts:
```diff
@@ -14,4 +14,15 @@ export class AppComponent {
+    console.log(this.listaTareas);
+  }
+
+  editar(index) {
```
Está cool :) Podrías ponerle a este método el nombre de **marcarTareaComoCompletada** cuando lo actualices de acuerdo a lo que hablamos

DentVega committed:
- Creacion del objeto Tarea
- Edicio de los metodos de creacion, eliminacion y edicion de una tarea
- modificacion del html para mostrar la informacion del objeto Tarea

DentVega committed:
- Correcciones

combimauri commented on src/app/app.component.html:
```diff
@@ -6,8 +6,16 @@ <h1>tareas</h1>
+         <a class="btn btn-red" (click)="eliminarTarea(indice)">
+           Eliminar
+         </a>
+         <button class="btn btn-red" (click)="marcarTareaComoCompletada(indice)">
```
Aquí podrías pasar directo la tarea en vez de mandar solo el índice

combimauri commented on src/app/app.component.html:
```diff
@@ -6,8 +6,16 @@ <h1>tareas</h1>
+      <li *ngFor="let tarea of listaTareas; index as indice">
+        <label [ngClass]="tarea.terminado ? 'strike' : 'null'">{{ tarea.detalle }}</label>
+        <label>
+         <a class="btn btn-red" (click)="eliminarTarea(indice)">
```
Creo que esto estilos no existen

combimauri commented on src/app/app.component.html:
```diff
@@ -6,8 +6,16 @@ <h1>tareas</h1>
-      <li *ngFor="let tarea of listaTareas">
-        <label>{{ tarea }}</label>
+      <li *ngFor="let tarea of listaTareas; index as indice">
+        <label [ngClass]="tarea.terminado ? 'strike' : 'null'">{{ tarea.detalle }}</label>
```
porfa, usa algo así:
```
[ngClass]="{ 'tarea-completada': tarea.completada }"
```
Estás asignado una clase null si es que no aplica la condición, esa clase no existe.

combimauri commented on src/app/Tarea.model.ts:
```diff
@@ -0,0 +1,9 @@
+export class Tarea {
```
El nombre de este archivo es incorrecto, deberías usar kebab case, debería llamarse: tarea.model.ts

combimauri commented on src/app/Tarea.model.ts:
```diff
@@ -0,0 +1,9 @@
+export class Tarea {
+  detalle: string;
+  terminado: boolean;
```
Este atributo debería ser: terminada o completada, en femenino

combimauri commented on src/app/app.component.css:
```diff
@@ -0,0 +1,3 @@
+.strike {
```
Mejor si usas un nombre como: task-completed o su equivalente en español

combimauri commented on src/app/app.component.ts:
```diff
@@ -7,11 +8,23 @@ import { Component } from '@angular/core';
+    tarea.terminado = true;
+    this.listaTareas[index] = tarea;
+  }
+
```
Elimina esta línea blanca

DentVega committed:
- Correcciones

combimauri commented on src/app/app.component.ts:
```diff
@@ -7,11 +8,21 @@ import { Component } from '@angular/core';
+    this.listaTareas.splice(indice, 1);
+  }
+
+  marcarTareaComoCompletada(tarea, indice) {
```
No es necesario que mandes el indice en este método

combimauri commented on src/app/app.component.ts:
```diff
@@ -7,11 +8,21 @@ import { Component } from '@angular/core';
+
+  marcarTareaComoCompletada(tarea, indice) {
+    tarea.completada = !tarea.completada;
+    this.listaTareas[indice] = tarea;
```
Esta línea de código es innecesaria

combimauri commented on src/app/app.component.html:
```diff
@@ -6,8 +6,16 @@ <h1>tareas</h1>
+          <input type="checkbox" class="todo-checkbox" (click)="marcarTareaComoCompletada(tarea, indice)"/>
+          <label [ngClass]="{ 'tarea-completada': tarea.completada }">{{ tarea.detalle }}</label>
+          <label>
+            <button class="btn btn-red" (click)="eliminarTarea(tarea)">
```
```suggestion
            <button class="btn btn-red" (click)="eliminarTarea(indice)">
```

DentVega committed:
- Correcciones

DentVega commented:
@mauricioarce corregido

combimauri reviewed (commented):
Buenísimo. Agrega las instrucciones al Readme y listo :)

DentVega committed:
- Añadido de las instrucciones de la segunda parte.

DentVega commented:
@mauricioarce  instrucciones de la segunda parte añadidas
